### PR TITLE
Fix for #54 - allow regexes when matching expected message text

### PR DESCRIPTION
**Pull request by brunns**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ On top of that, each case must comply to following types:
 | `env`           | `Optional[Dict[str, str]]={}`                          | Environmental variables to be provided inside of test run                                                       |
 | `parametrized`  | `Optional[List[Parameter]]=[]`\*                       | List of parameters, similar to [`@pytest.mark.parametrize`](https://docs.pytest.org/en/stable/parametrize.html) |
 | `skip`          | `str`                                                  | Expression evaluated with following globals set: `sys`, `os`, `pytest` and `platform`                           |
+| `regex`         | `str`                                                  | Allow regular expressions in comments to be matched against actual output.                                      |
```
Comment (sobolevn) [inline README.md:65]: Let's say that the default is `no`

Comment (brunns) [reply README.md:65]: It does indeed default to "no" - I'll mention that.

File: pytest_mypy_plugins/tests/test-regex_assertions.yml [hunk 2]
```diff
@@ -6,3 +6,11 @@
 
     reveal_type(a)  # N: Revealed type is "builtins.int"
     reveal_type(b)  # N: .*str.*
+
+- case: expected_message_regex_with_out
```
Comment (sobolevn) [inline pytest_mypy_plugins/tests/test-regex_assertions.yml:10]: We also need a test without `regex` attribute and with `regex: no` where we try to use regex outout

**Review by sobolevn — COMMENTED** [review]
Great work! Thank you! Is it really still "WIP"?

Please, also check out https://github.com/typeddjango/pytest-mypy-plugins/issues/45 
Maybe it is related / can be easily added.

File: pytest_mypy_plugins/tests/test-regex_assertions.yml (hunk 2 above)
Comment (brunns) [reply pytest_mypy_plugins/tests/test-regex_assertions.yml:10]: I'm not sure how to do negative tests - ie tests where we expect a failure fail. Are there any examples of this kind of test?

**Comment by brunns** [comment]
> Is it really still "WIP"?

I suppose it could be merged now, but there are a couple of things we _might_ want to add. Firstly, the template idea from #54, and secondly, some way of making a single expectation a regex rather than the whole test. What do you think?

Re #45, I think this PR works for that use case, though in a slightly different way.

File: pytest_mypy_plugins/tests/test-regex_assertions.yml (hunk 2 above)
Comment (brunns) [reply pytest_mypy_plugins/tests/test-regex_assertions.yml:10]: For example, if I added this test:

```yaml
- case: should_fail
  regex: no
  main: |
    a = 'hello'
    reveal_type(a)  # N: .*str.*
```

I'd expect the test to fail - but that fails the build. Is there a way to express that I _expect_ this to fail?

Comment (sobolevn) [reply pytest_mypy_plugins/tests/test-regex_assertions.yml:10]: Nope, see #46 

The easiest way for now is to add all failing tests into a single non-collectable file. And run it with `pytest` after the main test case. We can `grep` that `failed X tests`

Comment (brunns) [reply pytest_mypy_plugins/tests/test-regex_assertions.yml:10]: Can you approve me to run workflows? I have something which works locally, but I'd like to see if it works in CI.

**Comment by sobolevn** [comment]
> a single expectation a regex rather than the whole test

Maybe we can use `NR: .*regex.*`, where `R` suffix means `regex`?

**Comment by brunns** [comment]
I'm not sure the template idea belongs in this PR, so I think it's ready to go AFAIC.

File: pytest_mypy_plugins/tests/test-regex_assertions.shouldfail.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+- case: rexex_but_not_turned_on
+  main: |
+    a = 'hello'
+    reveal_type(a)  # N: .*str.*
+
+- case: rexex_but_turned_off
```
Comment (sobolevn) [inline pytest_mypy_plugins/tests/test-regex_assertions.shouldfail.yml:6]: Let's add one more test that should fail: with wrong regex. Something that does not match the output 👍

**Review by sobolevn — COMMENTED** [review]
Great work! One last test and we are good to go! 

Thanks a lot for your time and effort! 👍

**Comment by sobolevn** [comment]
Thanks a lot for your help!

**Pull request merged**